### PR TITLE
Adds Cucumber integration

### LINKED ISSUE #72: Is it possible to use gem with Cucumber + Capybara + RSpec?
### Discussed in https://github.com/donv/capybara-screenshot-diff/discussions/71

<div type='discussions-op-text'>

<sup>Originally posted by **gsguma** February 24, 2023</sup>
I'm trying to implement this gem using Cucumber + Capybara + RSpec, but it didn't work. Is it possible to use this gem in the cucumber steps?

I tried the following changes:

**env.rb:**
```
require 'capybara/screenshot/diff'

...

RSpec.configure do |config|
  config.include Capybara::Screenshot::Diff
end
```

**test_steps.rb:**
```
And(/^visit google$/) do
  visit 'https://google.com'

  screenshot 'test'
end
```

**Gemfile:**
```
  ...

  gem 'capybara-screenshot-diff'
  gem 'oily_png', platform: :ruby

  ...
```

And I'm getting the error `undefined method 'screenshot' for #<...+Capybara::Screenshot::Diff+Capybara::Screenshot::Os+Capybara::RSpecMatchers+Capybara::DSLRSpecProxyInstaller+Capybara::DSL+RSpec::Matchers+Cucumber::Glue::ProtoWorld+ ...> (NoMethodError)`

If this gem is cucumber compatible, do you have any idea that can help me?

Thank you in advance! </div>

### PR DESCRIPTION
**Pull request by pftg**
Adds simple Cucumber integration.

I also added a simple example project https://github.com/jetthoughts/capybara-screenshot-diff-cucumber-example to show how it should works for different cases

image

Closes #72

### CODE REVIEW

**Comment by pftg** [comment]
@UweKubosch please check

File: lib/capybara/screenshot/diff/test_methods.rb [hunk 1]
```diff
@@ -88,7 +100,9 @@ def assert_image_not_changed(caller, name, comparison)
 
           return unless result
 
-          "Screenshot does not match for '#{name}' #{comparison.error_message}\nat #{caller.join("\n")}"
+          result = +"Screenshot does not match for '#{name}' #{comparison.error_message}"
+          result += "\nat #{caller.join("\n")}" if backtrace
```
Comment (pftg) [inline lib/capybara/screenshot/diff/test_methods.rb:104]: @UweKubosch @donv, what do you think I'll remove to show the backtrace in the message and test the framework to show the backtrace instead?

File: README.md [hunk 2]
```diff
@@ -67,6 +67,21 @@ RSpec.configure do |config|
...
+Load Cucumber support by adding the following line (typically to your `features/support/env.rb` file):
+
+```ruby
+require 'capybara/screenshot/diff/cucumber'
+```
+
+And in the steps you can use:
+
+```ruby
+Then('I should not see any difference') do
```
Comment (donv) [inline README.md:81]: Maybe "I should not see any visual difference" ?

File: lib/capybara/screenshot/diff/test_methods.rb (hunk 1 above)
Comment (donv) [reply lib/capybara/screenshot/diff/test_methods.rb:104]: Yeah, if the test framework can show the stack trace, that would be better.

**Review by donv — APPROVED** [review]

**Pull request merged**